var/spack/repos/builtin/packages/hpctoolkit/package.py

```diff
@@ -154,3 +161,18 @@ def configure_args(self):
             args.append('--enable-all-static')
 
         return args
+
+    # Remove setenv of ROCM, HIP, etc from the module file.  Loading
+    # the hpctoolkit module is not relevant to building a GPU app and
+    # some variables (HIP_PATH) intefere with building the app.
+    def setup_run_environment(self, env):
+        keeplist = []
+        for elt in env.env_modifications:
+            if not (isinstance(elt, SetEnv)
+                    and (elt.name.find('ROCM') >= 0
+                         or elt.name.find('HIP') >= 0
+                         or elt.name.find('CUDA') >= 0)):
+                keeplist.append(elt)
```

Comment:
I would rather:
- Search for the variables you want to remove
- Explicitly unset those variables

In this way you won't remove actions added by callbacks before this one (which may be more difficult to track or debug) but you will nullify them with an `unset`. Is this approach viable in your case?